crates/rattler_shell/src/activation.rs

```diff
@@ -474,6 +474,28 @@ impl<T: Shell + Clone> Activator<T> {
         Ok(ActivationResult { script, path })
     }
 
+    /// Create a deactivation script for the environment.
+    /// This returns the deactivation script that unsets environment variables
+    /// and runs deactivation scripts.
+    pub fn deactivation(&self) -> Result<ActivationResult<T>, ActivationError> {
```

Comment:
How does this handle "previous" values of environment variables? How does conda handle that?

So say you have an environment variable `FOO` set to `1`. You then activate an environment which sets `FOO` to `2`. When you now deactivate, what should `FOO` be?

Reply:
Shall we do the same then?

Reply:
If we want to do the same thing we would need to "store" the previous env vars by putting them in a temporary place. 

I think the way that would work is by adding env vars that we overwrite with a prefix, likie `PIXI_ENV_{SHLVL}_{ENV_VAR_NAME}` where SHLVL is the "shell level" (ie a number of how many environments are already activated) and ENV_VAR_NAME is the name of the env var we are overwriting.

When unsetting env vars we can then check if there are any env vars with SHLVL-1 and the name, and instead of unsetting we can restore from there. 

Does that make sense, roughly, @mrswastik-robot? We can also double check how Conda implements this.

Reply:
I think that sounds good except that we should maybe pass in the existing environment as an argument to the activator instead of reading it from the system :)

Reply:
It could also be part of the `ActivationVariables` struct, maybe? I need to re-read the code to fresh up my memory. But basically the idea would be that the user can decide to pass in the existing env vars or override with something entirely custom.

Reply:
Almost. IMO we should not read from the system at all in these methods, and handle all reading of env vars outside of the functions here.